src/DoubleResetDetector.cpp

```diff
@@ -19,13 +19,24 @@ DoubleResetDetector::DoubleResetDetector(int timeout, int address) {
 	waitingForDoubleReset = false;
 }
 
+DoubleResetDetector::DoubleResetDetector(int timeout, int address, void (*callback)(bool waitState)) {
+	this->timeout = timeout*1000;
+	this->address = address;
+	doubleResetDetected = false;
+	waitingForDoubleReset = false;
+	this->callback = callback;
+}
+
 bool DoubleResetDetector::detectDoubleReset() {
 	doubleResetDetected = detectRecentlyResetFlag();
 	if (doubleResetDetected) {
 		clearRecentlyResetFlag();
 	} else {
 		setRecentlyResetFlag();
 		waitingForDoubleReset = true;
+		if (callback != nullptr) {
+			callback(waitingForDoubleReset);
+		}
 	}
```

Comment:
Since you have to call detectDoubleReset to trigger such a callback is it not just as easy to use the return value?   
My inspiration for doing this was being able to indicate to the user when pressing the reset button would cause a double reset.  I use the callback to make the LED flash in a particular way.

Reply:
I see, and I think that being able to indicate that it's 'live' is a nice feature to add.
However a simpler solution would be to expose the condition used in loop() as a new public method something like:

    bool DoubleResetDetector::timeToStop() {
        return waitingForDoubleReset && millis() > timeout;
    }

Then DoubleResetDetector doesn't need to know about the callback method, as you can use the response from `drd.detectDoubleReset()` to know that you're not a double reset but you've started waiting, and instead of calling `drd.loop()` in your example, you can have:

    if (drd.timeToStop()) {
        drd.stop();
        drd_callback(false);
    }

I think this would also make it easier to understand even without your callback.